src/features/import-map.js

```diff
@@ -4,46 +4,90 @@
  * <script type="systemjs-importmap">{}</script>
  * OR
  * <script type="systemjs-importmap" src=package.json></script>
- * 
- * Only supports loading the first import map
  */
 import { baseUrl as pageBaseUrl, parseImportMap, resolveImportMap } from '../common.js';
 import { systemJSPrototype } from '../system-core.js';
 
-var importMap, importMapPromise;
+var importMapPromise, fetchPromises = {};
+
+// Start downloading the import maps as soon as possible
 if (typeof document !== 'undefined') {
-  const scripts = document.getElementsByTagName('script');
-  for (let i = 0; i < scripts.length; i++) {
-    const script = scripts[i];
-    if (script.type !== 'systemjs-importmap')
-      continue;
-
-    if (!script.src) {
-      importMap = parseImportMap(JSON.parse(script.innerHTML), pageBaseUrl);
-    }
-    else {
-      importMapPromise = fetch(script.src)
+  const scriptsToDownload = document.querySelectorAll('script[type="systemjs-importmap"][src]');
```

Comment:
Instead of checking for `src` in a different querySelectorAll call, could we just do one query selector, that groups them all together into a single import maps promise.

This does mean that all resolutions will be blocked even if we already had some import maps inline, but that is the way the spec is going so that is the way it is.

Reply:
At what moment in time should the document order of the import maps be considered? I was thinking that it was the moment of the first import, which is not always the same moment as when systemjs is executed.

The reason I held onto the promises instead of just chaining the promises like you suggested below is because I was holding off on merging the import maps in document order until the moment that the first System.import occurred. This allows for someone to script tag the systemjs library, dynamically inject import maps afterwards, and then later do the first System.import. All while still having the document order of the import maps preserved.

Rereading the proto spec and readme, I'm not finding anything that clarifies the desired behavior. Do you know of any part of the spec or github issues that clarifies this?